Python docs improvements for 3.12 (#10192)
PipedreamHQ/pipedream: docs/docs/code/python/working-with-files/README.md

thumbnail: https://res.cloudinary.com/pipedreamin/image/upload/v1646763737/docs/icons/icons8-opened-folder_y60u9l.svg
---

# Working with the filesystem in Python

> You'll commonly need to work with files in a workflow, for example: downloading content from some service to upload to another. This doc explains how to work with files in Pipedream workflows and provides some sample code for common operations.

Reviewer: ```suggestion
You can work with files within a workflow. For instance, downloading content from some service to upload to another. Here are some sample code for common file operations.
```

Reviewer: Just simplifying a bit

[See this example](/code/python/http-requests/#downloading-a-file-to-the-tmp-directory) to learn how to download a file to `/tmp`.

## Uploading a file from `/tmp`

[See this example](/code/python/http-requests/#uploading-a-file-from-the-tmp-directory) to learn how to upload a file from `/tmp` in an HTTP request.

## Downloading a file, uploading it in another `multipart/form-data` request

```python
import requests
from requests_toolbelt.multipart.encoder import MultipartEncoder
import os

def handler(pd: "pipedream"):
  download_url = "https://example.com"
  upload_url = "http://httpbin.org/post"
  file_path = "/tmp/index.html"
  content_type = "text/html"

  # DOWNLOAD
  with requests.get(download_url, stream=True) as response:
      response.raise_for_status()
      with open(file_path, "wb") as file:
          for chunk in response.iter_content(chunk_size=8192):
              file.write(chunk)

  # UPLOAD
  multipart_data = MultipartEncoder(fields={
    'file': (os.path.basename(file_path), open(file_path, 'rb'), content_type)
  })
  response = requests.post(
    upload_url,
    data=multipart_data,
    headers={'Content-Type': multipart_data.content_type}
  )
```

> 

Reviewer: Perhaps consider adding a `::: warning` about that?

> 

Author: Added the `::: warning` in https://github.com/PipedreamHQ/pipedream/pull/10192/commits/89ee6a5cd79c9100e52b77b5ea6d9decb4ee6256
